### PR TITLE
Fix #2507 remove use of cancel task in supervisor

### LINKED ISSUE #2507: Eliminate cancel task
I think we went down a rabbit hole with canceling tasks. There's only [one call to cancel](https://github.com/radiasoft/sirepo/blob/0941297a19561e431d2372546d2f9d6f2bca192b/sirepo/job_supervisor.py#L909), which means we've isolated all the cases where cancel might be called. Tornado does not cancel. It does throw exceptions, which is related but different.

There are a few problems that we need to manage, but they are all manageable. For example, instead of agent_start, the supervisor sends to a task that manages the agent. If the agent dies, the agent task in the supervisor replies with "abort". This would eliminate the whole websocket ready mess. 

I don't anticipate working on this, but just wanted to document some thoughts -- thanks to Tom Cargill.

See also #2467

### LINKED ISSUE #6649: all coroutines need exception handlers
job_supervisor _run doesn't log all exceptions. It should. There are probably other cases.

### PR DESCRIPTION
**Pull request by robnagler**
- Fix #6649 catch exceptions in all asyncio tasks
- Improved logging of exceptions and error states
- Added 00_startup_test to cause fast fail on suite
- Removed some exception raises and replaced with return values
- Added job_supervisor.InvalidReq for better logging
- Op destruction sets is_destroyed which is checked as a stop sentinel for continued execution after an await
- Ensure all ops are destroyed in try/finally
- pkdebug_str in supervisor better protected, show DESTROYED for Op
- run_op is the sentinel for runSimulation instead of also compute_job_serial, which was unnecessary.
- Ensure replies always sent from supervisor APIs
- op.destroy handles all op state changes (was split with driver before)
- Removed blank_test (unnecessary)

### CODE REVIEW

File: sirepo/job_driver/__init__.py [hunk 1]
```diff
@@ -222,8 +224,14 @@ def _websocket_ready_timeout_cancel(self):
...
 
     async def _websocket_ready_timeout_handler(self):
-        pkdlog("{} timeout={}", self, self.cfg.agent_starting_secs)
-        await self._start_free_resources(caller="_websocket_ready_timeout_handler")
+        try:
+            if self._websocket_ready_timeout or self._websocket_ready.is_set():
+                pkdlog("ignore timeout {}, is ready", self)
+                return
+            pkdlog("{} timeout={}", self, self.cfg.agent_starting_secs)
+            await self._start_free_resources(caller="_websocket_ready_timeout_handler")
```
Comment (e-carlin) [inline sirepo/job_driver/__init__.py:232]: `TypeError: object NoneType can't be used in 'await' expression`

_start_free_resources is not async.

**Review by e-carlin — CHANGES_REQUESTED** [review]
Left some comments on the code.

One issue I noticed was that when the websocket_ready_timeout is executed the user doesn't get any information back. It just looks to them like their job is stuck simulating.

File: sirepo/job_driver/__init__.py (hunk 1 above)
Comment (robnagler) [reply sirepo/job_driver/__init__.py]: [e1c059c](https://github.com/radiasoft/sirepo/pull/6681/commits/e1c059c3b216f3c81fab82b5fe49385119bc44c9)

File: sirepo/job_driver/__init__.py [hunk 2]
```diff
@@ -103,10 +100,6 @@ def __init__(self, op):
     def destroy_op(self, op):
         """Clear our op and (possibly) free cpu slot"""
```
[the remark this answers is not in the document]
Comment (robnagler) [reply sirepo/job_driver/__init__.py]: [e1c059c](https://github.com/radiasoft/sirepo/pull/6681/commits/e1c059c3b216f3c81fab82b5fe49385119bc44c9)

File: sirepo/job_supervisor.py [hunk 3]
```diff
@@ -886,95 +911,89 @@ def _req_is_valid(self, req):
...
-                raise
+                pkdlog("exception={} op={} stack={}", e, op, pkdexc())
+                self.__db_update(
+                    error="Server error",
+                    internalError=op.internal_error or e,
+                    status=job.ERROR,
+                )
+                return False
+
+            if not _is_run_op(f"prepare_send success"):
```
[the remark this answers is not in the document]
Comment (robnagler) [reply sirepo/job_supervisor.py:950]: See the comment I added. [e1c059c](https://github.com/radiasoft/sirepo/pull/6681/commits/e1c059c3b216f3c81fab82b5fe49385119bc44c9)

File: sirepo/job_supervisor.py [hunk 4]
```diff
@@ -369,11 +380,14 @@ async def _receive_api_admJobs(self, req):
         return self._get_running_pending_jobs()
 
     async def _receive_api_beginSession(self, req):
-        c = self._create_op(job.OP_BEGIN_SESSION, req, job.SEQUENTIAL, "sequential")
+        c = None
         try:
-            await c.prepare_send()
+            c = self._create_op(job.OP_BEGIN_SESSION, req, job.SEQUENTIAL, "sequential")
+            if not await c.prepare_send():
```
[the remark this answers is not in the document]
Comment (robnagler) [reply sirepo/job_supervisor.py:387]: See comment added. [e1c059c](https://github.com/radiasoft/sirepo/pull/6681/commits/e1c059c3b216f3c81fab82b5fe49385119bc44c9)

File: sirepo/job_supervisor.py [hunk 5]
```diff
@@ -209,24 +218,27 @@ def destroy_op(self, op):
 
     @classmethod
     def get_compute_job_or_self(cls, req):
-        try:
-            j = req.content.computeJid
-        except AttributeError:
+        if not (j := req.content.get("computeJid")):
+            # no a compute job
```
[the remark this answers is not in the document]
Comment (robnagler) [reply sirepo/job_supervisor.py]: [e1c059c](https://github.com/radiasoft/sirepo/pull/6681/commits/e1c059c3b216f3c81fab82b5fe49385119bc44c9)

File: sirepo/job_driver/__init__.py [hunk 6]
```diff
@@ -222,8 +224,14 @@ def _websocket_ready_timeout_cancel(self):
             self._websocket_ready_timeout = None
 
     async def _websocket_ready_timeout_handler(self):
-        pkdlog("{} timeout={}", self, self.cfg.agent_starting_secs)
-        await self._start_free_resources(caller="_websocket_ready_timeout_handler")
+        try:
+            if self._websocket_ready_timeout or self._websocket_ready.is_set():
```
[the remark this answers is not in the document]
Comment (robnagler) [reply sirepo/job_driver/__init__.py]: [e1c059c](https://github.com/radiasoft/sirepo/pull/6681/commits/e1c059c3b216f3c81fab82b5fe49385119bc44c9)

File: sirepo/job_driver/__init__.py [hunk 7]
```diff
@@ -172,16 +165,25 @@ async def prepare_send(self, op):
...
+        if not await self._agent_ready(op):
+            return False
+        r = await self._slots_ready(op)
+        if r == job_supervisor.SlotAllocStatus.OP_IS_DESTROYED:
+            return False
+        if r == job_supervisor.SlotAllocStatus.HAD_TO_AWAIT:
+            if not await self._agent_ready(op):
+                return False
         self._prepared_sends[op.op_id] = op
+        return True
```
[the remark this answers is not in the document]
Comment (robnagler) [reply sirepo/job_driver/__init__.py:188]: [e1c059c](https://github.com/radiasoft/sirepo/pull/6681/commits/e1c059c3b216f3c81fab82b5fe49385119bc44c9)

File: sirepo/job_supervisor.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (e-carlin) [reply sirepo/job_supervisor.py:950]: Thanks. I was reading op.run_op not self.run_op. The comment helps.

**Review by e-carlin — APPROVED** [review]

**Pull request merged**